includes/Plugin_Context.php

```diff
@@ -13,5 +13,58 @@
  * @since n.e.x.t
  */
 class Plugin_Context {
-	// @TODO: To complete as part of issue #5
+
+	/**
+	 * Absolute path of the plugin main file.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	protected $main_file;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $main_file The absolute path to the plugin main file.
+	 */
+	public function __construct( $main_file ) {
+		$this->main_file = $main_file;
+	}
+
+	/**
+	 * Returns the plugin basename.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Plugin basename.
+	 */
+	public function basename() {
+		return plugin_basename( $this->main_file );
+	}
+
+	/**
+	 * Returns the absolute path for a relative path to the plugin directory.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $relative_path Optional. Relative path. Default '/'.
+	 * @return string Absolute path.
+	 */
+	public function path( $relative_path = '/' ) {
+		return plugin_dir_path( $this->main_file ) . ltrim( $relative_path, '/' );
```

Comment:
Should update this line with `return plugin_dir_path( WP_PLUGIN_CHECK_PLUGIN_DIR_PATH ) . $this->main_file . ltrim( $relative_path, '/' );` to get full relative path. If we use this and run the unit test it gives `Use of undefined constant WP_PLUGIN_CHECK_PLUGIN_DIR_PATH - assumed 'WP_PLUGIN_CHECK_PLUGIN_DIR_PATH' (this will throw an Error in a future version of PHP)` notice that needs to figure out.

@jjgrainger I'd like to get your feedback on this.

Reply:
Thanks @mukeshpanchal27 

I've addressed this in https://github.com/10up/plugin-check/pull/51. We're unable to use the `WP_PLUGIN_CHECK_PLUGIN_DIR_PATH` as it's created in the `plugin-check.php` file which is not bootstrapped as part of the tests (nor should it be).

The line here works correctly as the full path is expected to be passed in the constructor (see the [`plugin-check.php file here](https://github.com/10up/plugin-check/blob/trunk/plugin-check.php#L45)).

I've updated the tests based on the feedback in #51 which I believe to be correct.